pkg/vgmanager/vgmanager_controller.go

```diff
@@ -111,64 +160,55 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 		deviceClassMap[deviceClass.Name] = i
 	}
 
-	// filter out block devices
-	remainingValidDevices, delayedDevices, err := r.filterAvailableDevices(blockDevices)
+	_, found := deviceClassMap[volumeGroup.Name]
+
+	// create/extend VG and update lvmd config
+	err = r.addDevicesToVG(volumeGroup.Name, matchingDevices)
 	if err != nil {
-		_ = err
-	}
+		r.Log.Error(err, "failed to create/extend volume group", "VGName", volumeGroup.Name)
 
-	for _, deviceClass := range lvmCluster.Spec.DeviceClasses {
-		// ignore deviceClasses whose LabelSelector doesn't match this node
-		// NodeSelectorTerms.MatchExpressions are ORed
-		node := &corev1.Node{}
-		selectsNode, err := NodeSelectorMatchesNodeLabels(node, deviceClass.NodeSelector)
-		if err != nil {
-			r.Log.Error(err, "failed to match nodeSelector to node labels")
-			continue
-		}
-		if !(selectsNode && ToleratesTaints(deviceClass.Tolerations, node.Spec.Taints)) {
-			continue
-		}
-		_, found := deviceClassMap[deviceClass.Name]
 		if !found {
-			lvmdConfig.DeviceClasses = append(lvmdConfig.DeviceClasses, &lvmd.DeviceClass{
-				Name:        deviceClass.Name,
-				VolumeGroup: deviceClass.Name,
-				Default:     true,
-			})
+			status = "unavailable"
+			reason = "failed to create VG"
```

Comment:
`reason` for `unavailable` starts with small `f`, while that of `degraded` starts with capital `F`